backend/entityservice/views/project.py

```diff
@@ -151,20 +151,26 @@ def project_binaryclks_post(project_id):
                 # connexion has already read the data before our handler is called!
                 # https://github.com/zalando/connexion/issues/592
                 # stream = get_stream()
-                stream = BytesIO(request.data)
-                expected_bytes = binary_format(size).size * count
+                stream = iterable_to_stream(BytesIO(request.data))
+                binary_formatter = binary_format(size)
+                def entity_id_injector(filter_stream):
+                    for entity_id in range(count):
+                        yield binary_formatter.pack(entity_id, filter_stream.read(size))
+                data_with_ids = b''.join(entity_id_injector(stream))
```

Comment:
I want to check my understanding here. This takes the binary stream and converts it into a byte producing generator.
This generator uses the _user supplied_ `count` and reads `size` bytes (where `size` is also user supplied in the header) from the stream, packing the bytes using the binary format adding in an entity id as it goes.

- How do we handle the user supplying their own entity ids? I assume that is future work? Will they be optional/required?
- If this code path is here for the long haul, can we do this without making a full copy of the data in memory?

Reply:
I just wrote a quick fix to make this endpoint work with the new format.

Until connexion fixes the stream thing we cannot do much here.
There is also the problem, that the binary format as of now does not consider blocking information. 

I imagine, that in the far future, one day, someone will revisit the binary upload. (S)he will define a new binary format, implement that in the anonlink-client, and finally, will make this endpoint stunningly beautiful. 

For now, this is not a priority, though.